### ISSUE TITLE
Bug in showlog.cgi when returning to current log

### ISSUE THREAD

**Issue by mtnielsen** [NagiosEnterprises/nagioscore#906]
Nagios Core 4.4.9

Select Event Log from the navigation menu
Go back 1 log
Go forward 1 log
Error: "Bad Request"

---

As far as I can tell, the issue stems from

https://github.com/NagiosEnterprises/nagioscore/blob/master/cgi/showlog.c#L92

This bit adds an unconditional ? to the URL passed to display_nav_table(...)

When the URL is `/nagios/cgi-bin/showlog.cgi` everything is fine. When the URL is `/nagios/cgi-bin/showlog.cgi?` it explodes.

I'm not comfortable enough with C to fix it.

**Comment by ericloyd** [comment]
There seems to be a lot of these type of nav errors.  I think the intention was that you would always use the menu elements to go forward or backward, not the browser controls.  I'd love to see them all fixed.

**Comment by mtnielsen** [comment]
No browser controls involved. It's the Nagios navigation elements that produce the error.

![image](https://user-images.githubusercontent.com/5136850/231951628-4f26e3c7-d4ae-4359-9279-550997c61eac.png)

![image](https://user-images.githubusercontent.com/5136850/231951792-0459ace3-1652-4e2a-9186-2a7e1b615bfa.png)

![image](https://user-images.githubusercontent.com/5136850/231951827-1f4e1657-aac5-4698-b339-bc0d9a09eb44.png)

**Comment by mtnielsen** [comment]
No browser controls were used. It's all Nagios navigation buttons.

![image](https://user-images.githubusercontent.com/5136850/231952015-056c4f60-6d13-4e67-99d9-0df7f6788327.png)

![image](https://user-images.githubusercontent.com/5136850/231952035-304c2522-c314-4c30-8f2a-b458b2d672a5.png)

![image](https://user-images.githubusercontent.com/5136850/231952056-99660418-b721-49dc-ab00-aeb86c094970.png)

**Comment by sawolf** [comment]
Hi @mtnielsen, thanks for reaching out. I probably won't have this done for 4.4.11 (going out today), but I'll see if I can get this fixed for 4.4.12.

**Comment by sawolf** [comment]
I wasn't able to reproduce this issue (neither on 4.4.9 nor on the latest commits in master, and regardless of whether I used the navigation buttons or manually edited the URL). I also don't see a place in any of the CGIs where we'll intentionally return a 400 error (though I could have missed it).

My questions to you are:
 - what distribution are you using?
 - do you still see this problem after upgrading to the latest version of Nagios Core?
 - can you reproduce this from a fresh install?

**Comment by mtnielsen** [comment]
I did some follow up and it turns out nginx is removing the empty query string when passing along the request, which in turn causes a uri mismatch in httpd. 

I circumvented the proxy and couldn't reproduce the issue when connecting directly to the host.

I'm not sure what the correct behavior is supposed to be here. Is Nagios at fault for producing an empty query string? Is nginx at fault for removing the empty query string? Or is httpd at fault for complaining about a uri mismatch? Or is it perhaps Chrome's fault?

I'll have to investigate this tomorrow, but for the time being it doesn't look like Nagios' fault.
